docs/proposals/super-simple-discovery-api.md

```diff
@@ -0,0 +1,306 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Super Simple Discovery API
+
+## Overview
+
+It is surprisingly hard to figure out how to talk to a Kubernetes cluster.  Not only do clients need to know where to look on the network, they also need to identify the set of root certificates to trust when talking to that endpoint.
+
+This presents a set of problems:
+* It should be super easy for users to configure client systems with a minimum of effort `kubectl` or `kubeadm init` (or other client systems).
+  * Establishing this should be doable even in the face of nodes and master components booting out of order.
+  * We should have mechanisms that don't require users to ever have to manually manage certificate files.
+* Over the life of the cluster this information could change and client systems should be able to adapt.
+
+## ClusterInfo object
+
+First we define a set of information that identifies a cluster and how to talk to it.
+
+This includes:
+
+* A cluster serial number/identifier.
+  * In an HA world, API servers may come and go and it is necessary to make sure we are talking to the same cluster as we thought we were talking to.
+  * It looks like Federation already has this concept.  We need to back port that to non-federated clusters.
+* A set of addresses for finding the cluster.
```

Comment:
So my problem with this proposal is that in order to get this info... I need to know this info.  This proposal really needs to justify itself above just "get a kubeconfig file from a URL".